code/game/gamemodes/gang/gang_items.dm

```diff
@@ -75,6 +77,185 @@
+	if(!gang || (src in gang.bosses))
+		return
+	var/obj/item/device/gangtool/O = new /obj/item/device/gangtool/spare/lt(user.loc)
+	qdel(gangtool)
+	O.register_device(user)
+	gang.bosses_working = TRUE
+	user.put_in_hands(O)
+	if(spawn_msg)
+		to_chat(user, spawn_msg)
+
+
+
+/datum/gang_item/function/implant
+	name = "Influence-Enhancing Mindshield"
+	id = "mindshield"
+	cost = 10
+
+/datum/gang_item/function/implant/spawn_item(mob/living/carbon/user, datum/gang/gang, obj/item/device/vigilante_tool/VT)
+	var/obj/item/weapon/implant/mindshield/MS = new()
+	MS.implant(user, user, 0)
+	VT.vigilante_items -= /datum/gang_item/function/implant
+
+/datum/gang_item/function/implant/get_cost_display(mob/living/carbon/user, datum/gang/gang, obj/item/device/gangtool/gangtool)
+	return "([get_cost(user, gang, gangtool)] Influence)"
+
+
+
+/datum/gang_item/function/backup
+	name = "Create Gateway for Reinforcements"
+	id = "backup"
+	item_path = /obj/machinery/gang/backup
+
+/datum/gang_item/function/backup/can_see(mob/living/carbon/user, datum/gang/gang, obj/item/device/gangtool/gangtool)
+	if(gang.gateways >= GATEWAYMAX || !SSticker.mode.vigilantes)
+		return FALSE
+	return TRUE
+
+/datum/gang_item/function/backup/can_buy(mob/living/carbon/user, datum/gang/gang, obj/item/device/gangtool/gangtool)
+	if(gang.gateways >= GATEWAYMAX)
+		return FALSE
+	return TRUE
+
+/datum/gang_item/function/backup/spawn_item(mob/living/carbon/user, datum/gang/gang, obj/item/device/gangtool/gangtool)
+	gang.gateways++
+	var/obj/machinery/gang/backup/gate = new(get_turf(user), gang)
+	gate.G = gang
+
+/datum/gang_item/function/backup/purchase(mob/living/carbon/user, datum/gang/gang, obj/item/device/gangtool/gangtool)
+	var/area/usrarea = get_area(user.loc)
+	if(!(usrarea.type in gang.territory|gang.territory_new))
+		to_chat(user, "<span class='warning'>This device can only be spawned in territory controlled by your gang!</span>")
+		return FALSE
+	var/confirm_final = alert(user, "Your gang can only place ONE gateway, make sure it is in a well-secured location.", "Are you ready to place the gateway?", "This location is secure", "I should wait...")
+	if(confirm_final == "No")
+		return FALSE
+	return ..()
+
+/obj/machinery/gang/backup
+	name = "gang reinforcements gateway"
+	desc = "A gateway used by gangs to bring in muscle from other operations."
+	anchored = TRUE
+	density = TRUE
+	icon = 'icons/obj/machines/teleporter.dmi'
+	icon_state = "gang_teleporter_on"
+	var/datum/gang/G
+	var/list/mob/dead/observer/queue = list()
+	max_integrity = 400
+	obj_integrity = 400
+	var/final_guard = TRUE
+
+/obj/machinery/gang/backup/Initialize(mapload, datum/gang/gang)
+	. = ..()
+	G = gang
+	name = "[G] reinforcements gateway"
+	addtimer(CALLBACK(src, .proc/reinforce), max(10, (4500 - world.time)))
+	do_sparks(4, TRUE, src)
+
+/obj/machinery/gang/backup/Destroy(mapload, datum/gang/gang)
+	for(var/mob/M in contents)
+		qdel(M)
```

Comment:
https://github.com/tgstation/tgstation/blob/master/code/game/atoms_movable.dm#L206 if that's happening you need to figure out why